src/Microsoft.Azure.SignalR.AspNet/EndpointProvider/ServiceEndpointProvider.cs

```diff
@@ -49,13 +45,9 @@ internal class ServiceEndpointProvider : IServiceEndpointProvider
             _port = endpoint.Port;
             _algorithm = options.AccessTokenAlgorithm;
 
-            _provider = provider;
             Proxy = options.Proxy;
 
-            if (endpoint.AccessKey is AadAccessKey key)
-            {
-                _ = key.UpdateAccessKeyAsync(provider, loggerFactory ?? throw new ArgumentNullException(nameof(loggerFactory)));
-            }
+            accessKeyManager.AddServiceEndpoint(endpoint);
```

Comment:
`ServiceEndpointProvider` is created for every negotiate request. Why not add all the service endpoints in `ServiceEndpointManager`?